File: python_comparison/number.ts
const t0 = performance.now()

const N = 1024 * 1024 * 8
const arr = new Float32Array(N)
// manuall construct in JS
for (let i = 0; i < arr.length; ++i) {
  arr[i] = i
}
// scan and then sum
for (let i = 1; i < arr.length; ++i) {
  arr[i] = arr[i - 1] + arr[i]
}
let total = 0
for (let i = 0; i < arr.length; ++i) {
  total += arr[i]
}
console.log((performance.now() - t0) / 1e3, 'seconds')

File: python_comparison/number.py
import time

t0 = time.time()

N = 1024 * 1024 * 8
# let python cheat a bit with built ins
arr = list(range(N))
# scan and then sum
for i in range(1, len(arr)):
    arr[i] = arr[i - 1] + arr[i]
total = 0
for i in range(len(arr)):
    total += arr[i]
print(time.time() - t0, "seconds")
